### PR TITLE
Align types in API server's resource package.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

There was an awkward mix of Atoi -> int, int64 and uint64 with casting between them.  int is int64 on most platforms that we run on and resource IDs are unlikely to wrap an int64 but let's just use uint64 throughout so the code behaves the same on all systems.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: apiserver/pkg/storage/calico/resource.go [hunk 1]
```diff
@@ -428,15 +427,16 @@ func (rs *resourceStore) GuaranteedUpdate(
 		if err != nil {
 			return err
 		}
-		revInt, _ := strconv.Atoi(accessor.GetResourceVersion())
+		versionStr := accessor.GetResourceVersion()
+		var revInt uint64
+		if versionStr != "" {
+			revInt, err = strconv.ParseUint(versionStr, 10, 64)
+			return fmt.Errorf("failed to parse non-empty resource version %q key=%q: %w", versionStr, key, err)
```
Comment (fasaxc) [inline apiserver/pkg/storage/calico/resource.go:434]: This error check feels like the most contentious change here.  I'm going by the comment above: "Check for Revision no. If not set or less than the current version" implies that it should either be empty or a number we can parse.  Anything else feels like a problem.

File: apiserver/pkg/storage/calico/resource.go [hunk 2]
```diff
@@ -428,15 +427,16 @@ func (rs *resourceStore) GuaranteedUpdate(
 		if err != nil {
 			return err
 		}
-		revInt, _ := strconv.Atoi(accessor.GetResourceVersion())
+		versionStr := accessor.GetResourceVersion()
+		var revInt uint64
+		if versionStr != "" {
+			revInt, err = strconv.ParseUint(versionStr, 10, 64)
```
Comment (caseydavenport) [inline apiserver/pkg/storage/calico/resource.go:433]: I think we need to check if the err is non-nil here

**Review by caseydavenport — APPROVED** [review]
One missing err check but otherwise this LGTM

File: apiserver/pkg/storage/calico/resource.go (hunk 2 above)
Comment (fasaxc) [reply apiserver/pkg/storage/calico/resource.go:433]: Oops, yep, reworked the code and left it half baked

**Pull request merged**